### PR TITLE
Prune canonical attestations when head changes

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

Optimization 

**What does this PR do? Why is it needed?**

Prune attestations from canonical block under `forkchoceUpdateWithExecution`. This covers the edge case where your parent block becomes canonical at the start of the slot. In this case, you don't want to include its attestations. We added feedback from https://github.com/prysmaticlabs/prysm/pull/11935

**Which issues(s) does this PR fix?**

Follows up on #11764 

**Other notes for review**

### CODE REVIEW

**Review by potuz — COMMENTED** [review]
I do not see what's achieved in this PR. The changes in #11764 already take care of pruning only canonical blocks upon insertion, why moving this call to `receive_attestation` is an improvement? this moves logic to a critical part of the block which is second zero.

**Comment by rauljordan** [comment]
@terencechain agreed with Potuz here. 11764 is already merged, so should we close this one?

**Comment by terencechain** [comment]
@rauljordan this is very different. This solves the issue where a block is non-canonical the first time around and then later becomes canonical. 

@potuz and I still need to discuss this which we haven't found time

File: beacon-chain/sync/subscriber_beacon_blocks.go [hunk 1]
```diff
@@ -47,21 +45,3 @@ func (s *Service) beaconBlockSubscriber(ctx context.Context, msg proto.Message)
 	}
 	return err
 }
-
-// The input attestations are seen by the network, this deletes them from pool
-// so proposers don't include them in a block for the future.
-func (s *Service) deleteAttsInPool(atts []*ethpb.Attestation) error {
```
Comment (terencechain) [inline beacon-chain/sync/subscriber_beacon_blocks.go:53]: `deleteAttsInPool` is not used

**Review by potuz — APPROVED** [review]

**Pull request merged**